joblib/test/test_parallel.py

```diff
@@ -554,3 +555,28 @@ def test_parallel_with_interactively_defined_functions():
 def test_parallel_with_exhausted_iterator():
     exhausted_iterator = iter([])
     assert_equal(Parallel(n_jobs=2)(exhausted_iterator), [])
+
+
+def check_memmap(a):
+    if not isinstance(a, np.memmap):
+        raise TypeError('Expected np.memmap instance, got %r',
+                        type(a))
+    return a.copy()  # return a regular array instead of a memmap
```

Comment:
This was the cause of the previous failure under windows: passing a temporary memmap instance back to the master process prevents the deletion of the pool temp folder as long as. I changed the delete_folder method to tolerate this (rare case) warning the user.

Under POSIX, there is no problem: the folder is deleted while the memmap instances are still fully functional until they get garbage collected.